trap/src/main/java/com/cursorinsight/trap/TrapApplication.kt

```diff
@@ -54,13 +54,20 @@ open class TrapApplication : Application() {
          * @param application The global android Application instance.
          * @return The singletonTrapManager instance.
          */
-        fun initialize(application: Application): TrapManager {
-            if (TrapApplication.application.get() == null) {
-                TrapApplication.application = WeakReference(application)
-                trapManager = TrapManager.getInstance(application, getConfig(application))
-            }
-
-            return trapManager
+        fun initialize(application: Application) {
+            Thread {
```

Comment:
Why is this "Thread {}" needed here?

Reply:
Because Android does not allow to perform network requests on the main thread, and if the NetworkTrapConfigProvider is used it downloads the config here. 
An alternative can be to keep the original code and provide a separate example for the NetworkTrapConfigProvider use case.

Reply:
Thanks!